test/testutil/apps_mem.c

```diff
@@ -7,13 +7,24 @@
  * https://www.openssl.org/source/license.html
  */
 
+#include <stdlib.h>
 #include "apps.h"
+#include "../testutil.h"
 
 /* shim that avoids sucking in too much from apps/apps.c */
 
 void *app_malloc(size_t sz, const char *what)
 {
-    void *vp = OPENSSL_malloc(sz);
+    void *vp;
 
+    /*
+     * This isn't ideal but it is what the app's app_malloc() does on failure.
+     * Instead of exiting with a failure, abort() is called which makes sure
+     * that there will be a good stack trace for debugging purposes.
+     */
+    if (!TEST_ptr(vp = OPENSSL_malloc(sz))) {
+        TEST_info("Could not allocate %zu bytes for %s\n", sz, what);
+        abort();
```

Comment:
you could just do exit since that's what the app version does.

Reply:
@richsalz, please read the comment (lines 21-23) one more time

Reply:
I read it and disagree.  So be it.